spec/recipes/server_spec.rb

```diff
@@ -85,6 +85,41 @@
     include_examples 'allow weak hmacs'
   end
 
+  context 'with permit_tunnel set to true' do
+    cached(:chef_run) do
+      ChefSpec::ServerRunner.new do |node|
+        node.normal['ssh-hardening']['ssh']['server']['permit_tunnel'] = true
+      end.converge(described_recipe)
+    end
+  end
+
+  context 'with permit_tunnel at deafult value of false' do
+    cached(:chef_run) do
+      ChefSpec::ServerRunner.new do |node|
+        node.normal['ssh-hardening']['ssh']['server']['permit_tunnel'] = false
+      end.converge(described_recipe)
+    end
+  end
+
+  context 'with permit_tunnel set to a valid string' do
+    cached(:chef_run) do
+      ChefSpec::ServerRunner.new do |node|
+        node.normal['ssh-hardening']['ssh']['server']['permit_tunnel'] = 'ethernet'
+      end.converge(described_recipe)
+    end
+  end
+
+  context 'with permit_tunnel set to an invalid string' do
+    cached(:chef_run) do
+      ChefSpec::ServerRunner.new do |node|
+        node.normal['ssh-hardening']['ssh']['server']['permit_tunnel'] = 'IAmNotValid'
+      end.converge(described_recipe)
+    end
+    it 'abort the Chef run' do
+      expect { chef_run }.not_to raise_error
```

Comment:
shouldn't it be `expect { chef_run }.to raise_exception('Incorrect value for attribute node['ssh-hardening']['ssh']['server']['permit_tunnel']: must be boolean or a string as defined in the sshd_config man pages, you passed "IAmNotValid"')` ?